### PR TITLE
ci/cd: drop distro-specific packaging workflows

### PR DESCRIPTION
**Pull request by ylobankov**
Tarantool 3.0.0 is not going to be distributed as a set of packages for
specific distros. Instead, it is planned to have a few `deb` and `rpm`
packages with a statically compiled Tarantool binary inside, which will
work on almost any distro. Also, we drop building packages with GC32.
Now only GC64 packages will be shipped.

Static build packaging was added at https://github.com/tarantool/tarantool/pull/8771.

### CODE REVIEW

File: changelogs/unreleased/drop-distro-based-packages.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## feature/build
+
+* Drop building distro-based `deb` and `rpm` packages.
```
Comment (p7nov) [inline changelogs/unreleased/drop-distro-based-packages.md:3]:
```suggestion
* Dropped building system-specific `deb` and `rpm` Tarantool packages.
```

File: changelogs/unreleased/gc64-packages-only.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+## feature/build
+
+* Switch to GC64 packages only.
```
Comment (p7nov) [inline changelogs/unreleased/gc64-packages-only.md:3]:
```suggestion
* Dropped building Tarantool packages with GC32. Now only GC64 packages are shipped.
```

**Review by p7nov — APPROVED** [review]
Please see my suggestions on the changelog.

File: changelogs/unreleased/drop-distro-based-packages.md (hunk 1 above)
Comment (ylobankov) [reply changelogs/unreleased/drop-distro-based-packages.md]: Fixed.

File: changelogs/unreleased/gc64-packages-only.md (hunk 2 above)
Comment (ylobankov) [reply changelogs/unreleased/gc64-packages-only.md]: Fixed.

**Pull request merged**